scripts/consume_sitemaps.py

```diff
@@ -135,8 +158,21 @@ def fetch_sitemap_files(authority, last_updated_date, rsession, url, db_session)
                     else:
                         logging.error(f"Error fetching thing for {thing_fetcher.url}")
                         thing_id = thing_fetcher.thing_identifier()
-                        sqlmodel_database.mark_thing_not_found(db_session, thing_id, thing_fetcher.url)
+                        sqlmodel_database.mark_thing_not_found(
+                            db_session, thing_id, thing_fetcher.url
+                        )
                     thing_futures.remove(thing_fut)
+                    if len(thing_futures) < REFETCH_LENGTH:
```

Comment:
New future construction site to prefetch futures while we work on saving the other ones to the db.